spi/control-plane-adapter-spi/src/main/java/org/eclipse/tractusx/edc/spi/cp/adapter/service/AdapterTransferProcessService.java

```diff
@@ -31,4 +35,15 @@ public interface AdapterTransferProcessService {
      * @return The result containing the contract negotiation id
      */
     ServiceResult<ContractNegotiation> initiateEdrNegotiation(NegotiateEdrRequest request);
+
+    /**
+     * Return a {@link EndpointDataReference} associated with the transferProcessId in input
+     *
+     * @param transferProcessId The transferProcessId
+     * @return The result containing the {@link EndpointDataReference}
+     */
+    ServiceResult<EndpointDataReference> findByTransferProcessId(String transferProcessId);
+
+    ServiceResult<List<EndpointDataReferenceEntry>> findCacheEntries(String assetId, String agreementId);
```

Comment:
`findbyAssetAndAgreement`, or even just `findBy` would be a more consistent name